backend/src/main/java/gov/cdc/usds/simplereport/api/apiuser/UserAccountCreationController.java

```diff
@@ -133,9 +132,54 @@ public void enrollEmailMfa(@RequestBody EnrollMfaRequest requestBody, HttpServle
    * @param request contains session information about the user, including their Okta id.
    * @throws OktaAuthenticationFailureException if Okta cannot enroll the user in MFA.
    */
-  @GetMapping("/authenticator-qr")
-  public void getAuthQrCode(@RequestBody EnrollMfaRequest requestBody, HttpServletRequest request) {
-    // WIP: doesn't interact with Okta yet.
+  @PostMapping("/authenticator-qr")
+  public FactorAndQrCode getAuthQrCode(
+      @RequestBody EnrollMfaRequest requestBody, HttpServletRequest request)
+      throws OktaAuthenticationFailureException {
+    String userId = getUserId(request.getSession());
+    FactorAndQrCode factorData =
+        _oktaAuth.enrollAuthenticatorAppMfa(userId, requestBody.getUserInput());
+    request.getSession().setAttribute(FACTOR_ID_KEY, factorData.getFactorId());
+    return factorData;
+  }
+
+  /**
+   * Enrolls a security key for the user, returning data needed to finish activation.
+   *
+   * @param request contains session information about the user, including their Okta id.
+   * @return activation object with information to finish enrolling the security key, including a
+   *     challenge and user id.
+   * @throws OktaAuthenticationFailureException if the user is not recognized or Okta cannot enroll
+   *     their security key.
+   */
+  @PostMapping("/enroll-security-key-mfa")
```

Comment:
General nitpick: the URLs are reasonably self-explanatory, but it seems like it could be made more structured, and I would argue that it would be good if they were. So

* `/mfa/enroll/security-key`
* `/mfa/activate/security-key`
* `/mfa/enroll/totp`
* `/mfa/activate` (for all the ones that just require returning a code and you don't care which type of factor it is)

This would also enable having a handler for `/mfa/enroll/{type}` that dispatched on the type instead of having a different handler method for each type of MFA, but that might be an unnecessary luxury. 😄